File: java/src/test/java/com/testexercise/ebayframework/SearchResultsPage.java
package com.testexercise.ebayframework;

import java.util.*;

import org.openqa.selenium.*;

public class SearchResultsPage extends BasePage
{
    public SearchResultsPage(AutomationFramework af)
    {
        super(af);
    }
    
    private final String resultListId = "ListViewInner";
    private final String resultListItemCss = "#ListViewInner > li";
    private final String itemListingTypeCss = "div.pnl-b.frmt";
    
    public WebElement resultsList()
    {
        return af.driver().findElement(By.id(resultListId));
    }
    
    public void viewOnlyBuyItNowItems()
    {
        af.driver().
            findElement(By.cssSelector(itemListingTypeCss)).
            findElement(By.linkText("Buy It Now")).
            click();
    }
    
    public SearchItemWidget getSearchResultOnPage(int index)
    {
        List<WebElement> results = resultsList().
                findElements(By.cssSelector(resultListItemCss));
        
        return new SearchItemWidget(results.get(index));
    }
}

File: net/TestExerciseEbay/EbayPOMFramework/SearchResultsPage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using OpenQA.Selenium;

namespace TestExercise.EbayPOMFramework
{
    public class SearchResultsPage : BasePage
    {
        public SearchResultsPage(AutomationFramework af) : base(af) { }

        private const String resultListId = "ListViewInner";
        private const String resultListItemCss = "#ListViewInner > li";
        private const String itemListingTypeCss = "div.pnl-b.frmt";

        public IWebElement resultsList()
        {
            return af.Driver.FindElement(By.Id(resultListId));
        }

        public void ViewOnlyBuyItNowItems()
        {
            af.Driver.
                FindElement(By.CssSelector(itemListingTypeCss)).
                FindElement(By.LinkText("Buy It Now")).
                Click();
        }

        public SearchItemWidget GetSearchResultOnPage(int index)
        {
            IReadOnlyCollection<IWebElement> results = resultsList().
                    FindElements(By.CssSelector(resultListItemCss));

            return new SearchItemWidget(results.ElementAt(index));
        }
    }
}
